cmd/zdb/zdb.c

```diff
@@ -4424,21 +4353,31 @@ main(int argc, char **argv)
 	target = argv[0];
 
 	if (dump_opt['e']) {
+		importargs_t args = { 0 };
 		nvlist_t *cfg = NULL;
-		char *name = find_zpool(&target, &cfg, nsearch, searchdirs);
 
-		error = ENOENT;
-		if (name) {
-			if (dump_opt['C'] > 1) {
-				(void) printf("\nConfiguration for import:\n");
-				dump_nvlist(cfg, 8);
-			}
+		args.paths = nsearch;
+		args.path = searchdirs;
+		args.can_be_active = B_TRUE;
+
+		error = zpool_tryimport(g_zfs, target, &cfg, &args);
```

Comment:
It looks like the message emitted by zdb in this situation was `Matched %d pools - use pool GUID instead of pool name or pool name part of a dataset name to select pool`, but the message emitted by the zpool utility was `cannot import 't': more than one matching pool import by numeric ID instead`